crosslink/internal/tidy.go

```diff
@@ -0,0 +1,204 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crosslink
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	"go.uber.org/zap"
+	"golang.org/x/mod/modfile"
+)
+
+type graphNode struct {
+	file    *modfile.File
+	path    string
+	name    string
+	deps    []string
+	index   int
+	sccRoot int
+	onStack bool
+}
+
+func Tidy(rc RunConfig, outputPath string) error {
+	rc.Logger.Debug("crosslink run config", zap.Any("run_config", rc))
+
+	rootModule, err := identifyRootModule(rc.RootPath)
+	if err != nil {
+		return fmt.Errorf("failed to identify root module: %w", err)
+	}
+
+	// Read circular dependency allowlist
+
+	var allowCircular []string
+	if rc.AllowCircular != "" {
```

Comment:
I am not sure if we validate this, but it would be good to not only skip the modules listed here but to also very that these modules have a circular dependency (to make sure the allowlist is valid)